thirdparty/stb_image/stb_image.h

```diff
@@ -365,11 +366,6 @@ RECENT REVISION HISTORY:
 //    default this is set to (1 << 24), which is 16777216, but that's still
 //    very big.
 
-#ifdef __TINYC__
-#define STBI_NO_SIMD
-#define STBI_NO_THREAD_LOCALS
-#endif
```

Comment:
This was important for compiling with tcc on macos. I am not sure, if it is needed anymore 🤔, but imho it better stay for now.

Reply:
Just checked, it is still needed:
<img width="1005" alt="image" src="https://github.com/vlang/v/assets/26967/099dfdf4-342c-4195-999c-4b2672e7ed95">

... while with fd78504:
<img width="1167" alt="image" src="https://github.com/vlang/v/assets/26967/28502233-59c5-43ab-9f24-48d616455b2b">

Reply:
Nice, thanks for the review!

This would work. 
```
v -gc none -cc tcc -showcc -cflags -mfpu=neon-vfpv4 vlib/stbi/stbi_test.v
```

Is there a chance to intrinsically pass that flag on macos with tcc to use the lib in more comprehensively? Similar to what I do in the vab PR? Then the vab failure would fix itself without a change required there.

Reply:
No, it does not, it still needs the patches:
<img width="1186" alt="image" src="https://github.com/vlang/v/assets/26967/1b3818ac-5850-473e-bf84-4c073268ed38">

With them, it works with and without `-mfpu=neon-vfpv4`.

Reply:
Anyway, the benefit of tcc on all platforms, is not optimizations, but fast compilation, and I prefer to keep it working, rather than make the header lib more fiddly to deal with.

Reply:
I do not like the vab PR that makes it pass that specific option, since it is needed only for some targets (some Androids are not ARMs), but passed for all, and then we rely on the C compiler to ignore it, instead of error, which is brittle (such things change between compiler versions).

Reply:
Yes I wanted a quick fix a little too quick for this. Gonna update the vab PR when I have a free moment